chapter02/worksheets/solution2.5.2.6.scala

```diff
@@ -0,0 +1,41 @@
+/** 
+  Exercise 2.5.2.6
+
+  In a sorted integer array where no values are repeated, find all
+  pairs of values whose sum equals a given number 𝑛.
+  Use tail recursion.
+
+  A type signature and a sample test:
+
+  def pairs(goal: Int, xs: Array[Int]): Set[(Int, Int)] = ???
+
+  scala> pairs(10, Array(1, 2, 4, 5, 6, 8))
+  res0: Set[(Int, Int)] = Set((2,8), (4,6), (5,5))
+  */
+
+def pairsIntern(goal: Int, xs: Array[Int], res: Seq[(Int, Int)] = Seq((0, 0))): Seq[(Int, Int)] = { 
+  if (xs.isEmpty) res.reverse.tail
+  else
```

Comment:
We are using Scala 2, so you need braces after `else` if there is more than 1 expression.
```scala
if (...)...
else {
  val a = ...
  if (...)...
  else {
    val b = ...
  }
}
```

Reply:
Thanks, done.